Timesheet.gs

```diff
@@ -67,134 +74,152 @@ function checkOut (arg1, arg2, properties, lock) {return eventHandler_(EVENT_HAN
+    var initializeLog = false
+
+    if (arg1 !== undefined && arg1 instanceof Object && arg1.hasOwnProperty('authMode')) {
     
-      throw new Error('The third argument has to be one of the PropertiesServices')
+      // arg1 is an event so need to check authMode
+      if (arg1.authMode !== ScriptApp.AuthMode.NONE) { // LIMITED or FULL
+
+        userEmail = Session.getEffectiveUser().getEmail()
+        initializeLog = true
+      }
       
     } else {
-    
-      throw error
-    }
-  }
-  
-  try {
 
-    lock.hasLock()
+      // arg1 is not an event so assume we have sufficient auth to do the following
+      userEmail = Session.getEffectiveUser().getEmail()
+      initializeLog = true   
+    }
     
-  } catch (error) {
-  
-    if (error.message.indexOf('Cannot call method "hasLock" of undefined') !== -1) {
+    if (initializeLog) {
     
-      throw new Error('The fourth argument has to be one of the LockService')
+      Log_ = BBLog.getLog({
+        level:                DEBUG_LOG_LEVEL_, 
+        displayFunctionNames: DEBUG_LOG_DISPLAY_FUNCTION_NAMES_,
+      })
       
-    } else {
-    
-      throw error
+      Log_.info('Handling ' + config[0] + ' from ' + (userEmail || 'unknown email') + ' (' + SCRIPT_NAME + ' ' + SCRIPT_VERSION + ')')
     }
-  }
 
-  // Perform the main functionality
-  var originallyHasLock
-  try {
-
-    originallyHasLock = lock.hasLock()
-
-    // Perform any initial functions
-    config[0]()    
-
-    initialseEventHandler()
-    
-    var userEmail = Session.getEffectiveUser().getEmail()
-    Log_.info('Handling ' + config[1] + ' from ' + (userEmail || 'unknown email') + ' (' + SCRIPT_NAME + ' ' + SCRIPT_VERSION + ')')
-    
     // Call the main function
-    return config[3](arg1, arg2)
+    return config[2](arg1)
     
   } catch (error) {
   
-    Assert.handleError(error, config[2], Log_)
-    
-  } finally {
-  
-    if (!originallyHasLock) {
-      lock.releaseLock()
+    var handleError = Assert.HandleError.DISPLAY_FULL
+
+    if (!PRODUCTION_VERSION_) {
+      handleError = Assert.HandleError.THROW
     }
-  }
-  
-  return
-  
-  // Private Functions
-  // -----------------
-
-  /**
-   * Initialise the event handling
-   */
- 
-  function initialseEventHandler() {
-      
-    var userEmail = Session.getEffectiveUser().getEmail()
 
-    Assert.init({
-      handleError:    HANDLE_ERROR_, 
+    var assertConfig = {
+      error:          error,
+      userMessage:    config[1],
+      log:            Log_,
+      handleError:    handleError, 
       sendErrorEmail: SEND_ERROR_EMAIL_, 
-      emailAddress:   ADMIN_EMAIL_ADDRESS_ + ',' + userEmail,
+      emailAddress:   ADMIN_EMAIL_ADDRESS_,
       scriptName:     SCRIPT_NAME,
       scriptVersion:  SCRIPT_VERSION, 
-    })
+    }
 
-    if (PRODUCTION_VERSION_) {
+    Assert.handleError(assertConfig) 
+  }
+  
+} // eventHandler_()
+
+// Private event handlers
+// ----------------------
+
+function onInstall_(event){
```

Comment:
You can delete this function. It is only used for Add-ons.

Reply:
Ah, I didnt think this was used, but left in for completeness of the template, will remove.